### PR TITLE
assert: adding provision to enable asserts for all series

### PR DESCRIPTION
**Pull request by ikmdani**
These commits has changes necessary to make use of the present STM32Cube
hal & ll asserts for all series of stm32 platforms. Also the unused files has been
removed. Documentation updated.

### CODE REVIEW

**Comment by ikmdani** [comment]
@erwango, for your review. I could not add a reviewer here.

**Comment by erwango** [comment]
@ABOSTM We'll need to update update_serie script to take care of this.
(unless you have another suggestion for integration)

**Comment by ABOSTM** [comment]
> @ABOSTM We'll need to update update_serie script to take care of this.
> (unless you have another suggestion for integration)

@ikmdani 
I am not keen for such modifications (editing files stm32_assert.h ) of STM32cubes. And I wonder why we cannot simply rename stm32_assert_template.h to stm32_assert.h? I guess there is conflict between definitions in stm32f1xx_hal_conf.h and stm32_assert.h. HAL and LL are made to work together, so I think conflict should not happens. 
Can you give details of conflict you faced (if any) ? Which source is compiled, where is the conflict, ... ? Thanks

**Comment by ikmdani** [comment]
> I am not keen for such modifications (editing files stm32_assert.h ) of STM32cubes. And I wonder why we cannot simply rename stm32_assert_template.h to stm32_assert.h?

`stm32_assert_template.h` can be rename to `stm32_assert.h`, but then what i proposed in `stm32_assert.h` is needed. @erwango and I discussed here https://github.com/STM32-OSF/zephyr_hal_stm32/pull/1#discussion_r652481210.

> I guess there is conflict between definitions in stm32f1xx_hal_conf.h and stm32_assert.h

Are you pointing at this?
```
-void assert_failed(uint8_t* file, uint32_t line)
+void assert_failed(uint8_t *file, uint32_t line)
```

> Can you give details of conflict you faced (if any) ? Which source is compiled, where is the conflict, ... ?

1. The contents of `./stm32cube/stm32yyxx/drivers/include/stm32_assert_template.h` are already present at the end of `./stm32cube/stm32yyxx/drivers/include/stm32yyxx_hal_conf.h` files in all series.
2. `stm32_assert.h` may not be required at all. Since many of the ll files include `stm32_assert.h` when USE_FULL_ASSERT symbol is defined and keeping in mind that the stm32cube drivers shall not be changed, this is my suggestion.
Inclusion of `stm32_assert.h` in the ll drivers for L4 only shown below:

```
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_pka.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_tim.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_rng.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_lptim.c:28:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_utils.c:25:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_i2c.c:25:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_exti.c:24:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_lpuart.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_dma2d.c:25:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_opamp.c:25:  #include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_rtc.c:25:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_adc.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_dma.c:25:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_dac.c:26:  #include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_rcc.c:24:  #include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_usart.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_swpmi.c:25:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_crc.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_spi.c:26:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_comp.c:25:#include "stm32_assert.h"
./stm32cube/stm32l4xx/drivers/src/stm32l4xx_ll_gpio.c:25:#include "stm32_assert.h"
```

Pls suggest.

**Comment by ABOSTM** [comment]
@ikmdani ,
I don't have yet access to your discussion you pointed with @erwango,
but my point is the following:
theoretically, when compiling STM32 LL source file, there should be include of "stm32_assert.h" only (not stm32yyxx_hal_conf.h)
And reverse, when compiling STM32 HAL source there should be include of "stm32yyxx_hal_conf.h" only (not "stm32_assert.h").
So my suggestion is to simply rename "stm32_assert_template.h" to "stm32_assert.h". Then is there an issue doing so ?

**Comment by ABOSTM** [comment]
@ikmdani,
Ok I got the point: because all compilation switches: USE_HAL_DRIVER , USE_FULL_LL_DRIVER  and  USE_FULL_ASSERT
are enabled, then both `stm32_assert.h` and `stm32yyxx_hal_conf.h` will be included while compiling ll source files.
So let's continue with your implementation.

File: stm32cube/stm32l1xx/drivers/include/stm32_assert.h [hunk 1]
```diff
@@ -0,0 +1 @@
+#include <stm32l1xx_hal_conf.h>
```
Comment (ABOSTM) [inline stm32cube/stm32l1xx/drivers/include/stm32_assert.h:1]: Missing copyright and license information

File: README.rst [hunk 2]
```diff
@@ -82,6 +82,16 @@ driver Kconfig file should include the following entry
 	select USE_STM32_LL_I2C
 
 
+Similarlar to Zephyr, STM32Cube HAL and LL APIs embed asserts which are
```
Comment (ABOSTM) [inline README.rst:85]: typo: similarly

**Review by ABOSTM — CHANGES_REQUESTED** [review]

**Review by ABOSTM — APPROVED** [review]
LGTM

**Review by erwango — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**